description: prevent clang from printing unknown-warning warning. Also remove --rtlib=libgcc which won't be available until we have clang 13.
author: Wengling Chen <feiyu2817@gmail.com>

---
 build/config/android/BUILD.gn  |    8 --------
 build/config/compiler/BUILD.gn |    3 +++
 2 files changed, 3 insertions(+), 8 deletions(-)

--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -1598,6 +1598,9 @@ config("default_warnings") {
         # Flags NaCl (Clang 3.7) and Xcode 9.2 (Clang clang-900.0.39.2) do not
         # recognize.
         cflags += [
+          # Ignore unknown warning warnings from clang
+          "-Wno-unknown-warning-option",
+
           # An ABI compat warning we don't care about, https://crbug.com/1102157
           # TODO(thakis): Push this to the (few) targets that need it,
           # instead of having a global flag.
--- a/build/config/android/BUILD.gn
+++ b/build/config/android/BUILD.gn
@@ -51,14 +51,6 @@ config("compiler") {
     "-Wl,--exclude-libs=libvpx_assembly_arm.a",
   ]
 
-  # TODO(crbug.com/1184398): Move to compiler-rt when we are ready.
-  ldflags += [ "--rtlib=libgcc" ]
-  if (current_cpu == "arm64") {
-    # For outline atomics on AArch64 (can't pass this unconditionally
-    # due to unused flag warning on other targets).
-    cflags += [ "--rtlib=libgcc" ]
-  }
-
   # $compile_api_level corresponds to the API level used for the sysroot path
   # calculation in //build/config/android/config.gni
   if (android_64bit_target_cpu) {
